[ty] give Claude better test-running instructions (#22784)
astral-sh/ruff: CLAUDE.md

cargo nextest run -p ty_python_semantic
```

Run a single mdtest file:

```sh
cargo nextest run -p ty_python_semantic --test mdtest -- mdtest::<path/to/mdtest_file.md>
```

> Run a specific mdtest within a file (use a substring of the Markdown header text) -- this is usually not needed:

Reviewer: It might be more helpful to say when it is needed
```suggestion
Run a specific mdtest within a file (use a substring of the Markdown header text) -- Only use this when debugging a single test:
```
